Classes/EventListener/Linkvalidator/BeforeRecordIsAnalyzedEventListener.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+namespace T3docs\Examples\EventListener\Linkvalidator;
+
+use TYPO3\CMS\Core\DataHandling\SoftReference\SoftReferenceParserFactory;
+use TYPO3\CMS\Linkvalidator\Event\BeforeRecordIsAnalyzedEvent;
+use TYPO3\CMS\Linkvalidator\Repository\BrokenLinkRepository;
+
+final class BeforeRecordIsAnalyzedEventListener
```

Comment:
We should rename the class to something meaningful, like `CheckExternalLinksToLocalPages`.

Reply:
good idea